app/src/main/java/com/example/tasky/MainActivity.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.tasky
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.delay
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            setContent {
+                MyApp()
+            }
+        }
+    }
+}
+
+@Composable
+fun SplashScreen(onSplashComplete: () -> Unit) {
+        LaunchedEffect(key1 = true) {
+        delay(2000) // Delay for 2 seconds
+        onSplashComplete() // Notify that splash is complete
+    }
+    Box(
+        modifier = Modifier
+            .fillMaxSize() // Fill the entire screen
+            .background(colorResource(id = R.color.primary_color)), // Set the background color
+        contentAlignment = Alignment.Center
+    ) {
+        Image(
+            painter = painterResource(id = R.drawable.ic_tasky_logo),
+            contentDescription = null,
+            modifier = Modifier.size(200.dp)
+        )
+    }
+}
```

Comment:
I highly recommend to not create a separate splash screen. Android already has its own splash screen and that can lead to 2 splash screens that will be shown (exactly what happens in your video demo). Instead, you can use the official splash screen API which also lets you show the splash screen while checking if you're still logged in or not.

I explain it in this video: https://www.youtube.com/watch?v=Loo4i5IrZ4Y

Reply:
Thank you for the feedback. FYI - there is no more setKeepVisibleCondition -> It is now: 
setKeepOnScreenCondition.